### PR TITLE
alpaca: update apis

### PR DESCRIPTION
**Pull request by sc0Vu**
```BASH
$ p alpaca fetchOrderBook BTC/USD 
Python v3.11.4
CCXT v4.1.52
alpaca.fetchOrderBook(BTC/USD,1)
{'asks': [[36296.496, 0.28008],
          [36304.188, 0.832765],
          [36309.4, 0.5526],
          [36329.85, 1.12071],
          [36383.74, 1.4008],
          [36474.5, 1.654],
          [36666.461, 1.9624],
          [37000.0, 0.000275265],
          [37033.7, 2.2241],
          [38000.0, 0.000269864],
          [39587.0, 0.91816]],
 'bids': [[36259.7, 0.2774],
          [36248.02, 0.550762],
          [36246.59, 0.837],
          [36220.7, 1.113],
          [36174.81, 1.3848],
          [36067.4, 1.68328],
          [35889.7, 1.9562],
          [35562.0, 0.000281199],
          [35524.912, 2.2255],
          [34864.0, 0.000286828],
          [34180.0, 0.000292568],
          [33509.0, 0.000298427],
          [33000.0, 0.015],
          [32851.0, 0.000273654],
          [32206.0, 0.000310501],
          [31574.0, 0.000316716],
          [30954.0, 0.00032306],
          [30347.0, 0.000329521],
          [29751.0, 0.000336123],
          [29167.0, 0.000342853],
          [28595.0, 0.000349711],
          [28034.0, 0.000356709],
          [27484.0, 0.000363848],
          [26945.0, 0.000371126],
          [26416.0, 0.000378558],
          [26000.0, 0.099010991],
          [25898.0, 0.00038613],
          [25390.0, 0.000393855],
          [24813.895781637, 1.0],
          [24571.0, 1.0],
          [24000.0, 0.17],
          [19.851116625, 2.0],
          [9.925558312, 1.0],
          [1.0, 10.0],
          [0.149850149, 22.0]],
 'datetime': '2023-11-15T12:51:13.033Z',
 'nonce': None,
 'symbol': 'BTC/USD',
 'timestamp': 1700052673033}

$ p alpaca fetchTrades BTC/USD 1700002673033 1
Python v3.11.4
CCXT v4.1.52
alpaca.fetchTrades(BTC/USD,1700002673033,1)
[{'amount': 0.0004,
  'cost': 14.242214,
  'datetime': '2023-11-14T22:59:09.700Z',
  'fee': None,
  'fees': [],
  'id': '4281632672403637280',
  'info': {'i': '4281632672403637280',
           'p': '35605.535',
           's': '0.0004',
           't': '2023-11-14T22:59:09.700256623Z',
           'tks': 'S'},
  'order': None,
  'price': 35605.535,
  'side': 'sell',
  'symbol': 'BTC/USD',
  'takerOrMaker': 'taker',
  'timestamp': 1700002749700,
  'type': None}]

$ p alpaca fetchTrades BTC/USD 1700002673033 1 '{"method":"marketPublicGetV1beta3CryptoLocLatestTrades"}' 
Python v3.11.4
CCXT v4.1.52
alpaca.fetchTrades(BTC/USD,1700002673033,1,{'method': 'marketPublicGetV1beta3CryptoLocLatestTrades'})
[{'amount': 2.8184e-05,
  'cost': 1.0226169824,
  'datetime': '2023-11-15T12:47:36.588Z',
  'fee': None,
  'fees': [],
  'id': '7463763257731006751',
  'info': {'i': '7463763257731006751',
           'p': '36283.6',
           's': '0.000028184',
           't': '2023-11-15T12:47:36.588930217Z',
           'tks': 'B'},
  'order': None,
  'price': 36283.6,
  'side': 'buy',
  'symbol': 'BTC/USD',
  'takerOrMaker': 'taker',
  'timestamp': 1700052456588,
  'type': None}]


$ p alpaca fetchOHLCV BTC/USD 1m 1600000673033 1     
Python v3.11.4
CCXT v4.1.52
alpaca.fetchOHLCV(BTC/USD,1m,1600000673033,1)
[[1609480800000, 29255.71, 29283.39, 29237.45, 29257.99, 0.815545]]

$ p alpaca fetchOHLCV BTC/USD 1m 1600000673033 1 '{"method":"marketPublicGetV1beta3CryptoLocLatestBars"}' 
Python v3.11.4
CCXT v4.1.52
alpaca.fetchOHLCV(BTC/USD,1m,1600000673033,1,{'method': 'marketPublicGetV1beta3CryptoLocLatestBars'})
[[1700053260000, 36206.865, 36206.865, 36206.865, 36206.865, 0.0]]
```

### CODE REVIEW

File: ts/src/alpaca.ts [hunk 1]
```diff
@@ -364,18 +443,23 @@ export default class alpaca extends Exchange {
...
         await this.loadMarkets ();
         const market = this.market (symbol);
         const id = market['id'];
+        const loc = this.safeString (params, 'loc', 'us');
         const request = {
             'symbols': id,
+            'loc': loc,
         };
-        const response = await this.cryptoPublicGetCryptoLatestOrderbooks (this.extend (request, params));
+        params = this.omit (params, 'loc');
```
Comment (carlosmiei) [inline ts/src/alpaca.ts:461]: we don't need this omit here

**Comment by carlosmiei** [comment]
@sc0Vu did you test the order-related methods?

File: ts/src/alpaca.ts (hunk 1 above)
Comment (sc0Vu) [reply ts/src/alpaca.ts:461]: updated

File: ts/src/alpaca.ts [hunk 2]
```diff
@@ -552,7 +668,7 @@ export default class alpaca extends Exchange {
         const clientOrderId = this.safeString (params, 'clientOrderId', defaultClientId);
         request['client_order_id'] = clientOrderId;
         params = this.omit (params, [ 'clientOrderId' ]);
-        const order = await this.privatePostOrders (this.extend (request, params));
+        const order = await this.traderPrivatePostV2Orders (this.extend (request, params));
```
Comment (carlosmiei) [inline ts/src/alpaca.ts:671]: @sc0Vu if I try to place an order in the paper trading mode I get this error, is this expected?
```
✗ n alpaca createOrder "BTC/USD" market buy 1 5 --sandbox             
2023-11-16T12:40:32.255Z
Node.js: v18.18.0
CCXT v4.1.54
alpaca.createOrder (BTC/USD, market, buy, 1, 5)
PermissionDenied alpaca {"message": "forbidden."}

---------------------------------------------------
[PermissionDenied] alpaca {"message": "forbidden."}

    at throwExactlyMatchedException  Users/cjg/Git/ccxt13/ccxt/js/src/base/Exchange.js:3309  
    at handleErrors                  Users/cjg/Git/ccxt13/ccxt/js/src/alpaca.js:831          
    at                               Users/cjg/Git/ccxt13/ccxt/js/src/base/Exchange.js:782   
    at processTicksAndRejections     node:internal/process/task_queues:95                    
    at fetch2                        Users/cjg/Git/ccxt13/ccxt/js/src/base/Exchange.js:2885  
    at request                       Users/cjg/Git/ccxt13/ccxt/js/src/base/Exchange.js:2888  
    at createOrder                   Users/cjg/Git/ccxt13/ccxt/js/src/alpaca.js:552          
    at async run                     Users/cjg/Git/ccxt13/ccxt/examples/js/cli.js:319        

alpaca {"message": "forbidden."}
```

Comment (sc0Vu) [reply ts/src/alpaca.ts:671]: let me checked......

Comment (sc0Vu) [reply ts/src/alpaca.ts:671]: @carlosmiei I used production key, can you create order in production api?

**Pull request merged**